alibi_detect/cd/classifier.py

```diff
@@ -147,12 +148,12 @@ def __init__(
         self._set_config(locals())
 
         backend = backend.lower()
-        if (backend == 'tensorflow' and not has_tensorflow) or (backend == 'pytorch' and not has_pytorch) or \
-                (backend == 'sklearn' and not has_sklearn):
-            raise ImportError(f'{backend} not installed. Cannot initialize and run the '
-                              f'ClassifierDrift detector with {backend} backend.')
-        elif backend not in ['tensorflow', 'pytorch', 'sklearn']:
-            raise NotImplementedError(f"{backend} not implemented. Use 'tensorflow', 'pytorch' or 'sklearn' instead.")
+        BackendValidator(
+            backend_options={'tensorflow': ['tensorflow'],
+                             'pytorch': ['pytorch'],
+                             'sklearn': ['sklearn']},
```

Comment:
Similarly `'sklearn': ['scikit-learn']`? Although I tested this and it seems they aliased `sklearn` to pull in `scikit-learn`.

Reply:
In fact, `scikit-learn` is a core dependency, not sure if this check is even here.

Reply:
I copied the functionality from [here](https://github.com/SeldonIO/alibi-detect/blob/6e4f5e0c86e562d0a92bb48d0a9690de0606a37b/alibi_detect/cd/classifier.py#L150) and [here](https://github.com/SeldonIO/alibi-detect/blob/6e4f5e0c86e562d0a92bb48d0a9690de0606a37b/alibi_detect/utils/frameworks.py#L15). But yeah, I don't think it makes sense to have the full import check so I've just set `HAS_BACKEND['sklearnn] = True`. I think It is simpler just to keep the general logic in place even though it's redundant. Although happy to change?

Reply:
Just that [here](https://github.com/SeldonIO/alibi-detect/blob/6e4f5e0c86e562d0a92bb48d0a9690de0606a37b/alibi_detect/utils/frameworks.py#L15) we try and import `sklearn` and then set `has_sklearn` conditional on the result which is redundant as `sklearn` should always be there!